torchtune/modules/transformer.py

```diff
@@ -246,3 +247,152 @@ def forward(
         # shape: [b, s, out_dim] - out_dim is usually the vocab size
         output = self.output(h).float()
         return output
+
+
+class TiedEmbeddingTransformerDecoder(nn.Module):
```

Comment:
Now that this is in transformer.py as a core module we need to add a unit test for it. I don't think we need to block this PR on it, so filed #1241 as a follow-up.